Automattic/wp-calypso
# Store: Move products fetching / listing to the data layer

This PR moves the products fetching code from thunks to the data layer, similar to what was done in #16317. With the wpcom-http data layer, we'll get automatic retries for GET requests for free. 

It also adds some error handling to the product list page if we are unable to load the products correctly after those retries. Previously, nothing would happen -- now the user will get a message on the page instead of the list table.

I am committing these handlers to the old file location. #16444 moves them.

To Test:
* Run `npm run test-client client/extensions/woocommerce/state` and make sure all tests pass.
* Go to `http://calypso.localhost:3000/store/products/:site` and make sure that your products load correctly. Page between products as well.
* To test the error message, you can temporarily de-activate the `wc-api-dev` plugin.
